presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -259,6 +259,18 @@ public boolean schemaExists(Session session, CatalogSchemaName schema)
         return ImmutableList.copyOf(schemaNames.build());
     }
 
+    @Override
+    public String canonicalizeName(Session session, String catalogName, String identifier, boolean delimited)
+    {
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, catalogName);
+
+        if (catalog.isPresent()) {
+            return catalog.get().getMetadata().canonicalize(identifier, delimited);
+        }
+
+        return identifier.toLowerCase(ENGLISH);
```

Comment:
Have you run into any usages that fall through to this case?

Reply:
Nope, just a hook till we handle at all the layers